docs/proposals/edge-config-update.md

```diff
@@ -0,0 +1,212 @@
+---
+title: Support update EdgeCore config
+authors:
+- "@Shelley-BaoYue"
+  approvers:
+  creation-date: 2024-11-30
+  last-updated: 2024-11-30
+  status: implementable
+---
+
+# Support update EdgeCore config
+
+## Motivation 
+
+In many scenarios, being able to update the EdgeCore configuration files for edge nodes directly from the cloud provides significant convenience compared to logging into each edge node for manual updates. This is especially beneficial for batch operations, where managing multiple nodes at once can greatly enhance efficiency and streamline the operational process.
+
+### Goals
+
+- Provide APIs for creating config update tasks from cloud.
+- Support update edge config on multi edge nodes or nodegroup.
+- Support full or partial field updates to the edge configuration.
+- Support task rollback at edge. 
+
+## Design Details
+
+We propose using Kubernetes CRDs to describe ConfigUpdate metadata and a controller to synchronize task between edge and cloud.
+
+<img src="../images/edge-node-tasks/config-update-task.png">
+
+
+### ConfigUpdateController
+
+`configUpdateController` is responsible for dealing with `ConfigUpdateJob`
+
+- Use List-Watch mechanism to monitor `ConfigUpdateJob` CRD resources, receiving events and store it in local cache using map.
+- Use K8s informer to get node list according to NodeName or LabelSelector specified in CR, and filter out nodes that don't meet requirements (for example, edge node with NotReady status)
+- Check whether the job is completed. 
+
+### Tasks Manager
+
+TaskManager starts several goroutine called `upstream` controller, `downstream` controller and other task controllers. Task manager will reuse the design from the node task, please refer to https://github.com/kubeedge/kubeedge/blob/master/docs/proposals/edge-node-tasks-design.md for more details.
+
+### EdgeHub
+
+- Check the received config-update task.  
+
+### Keadm
+
+Add n new subcommand for keadm, `keadm update-config`, to process this task. keadm update-config will do: 
+
+- use `/etc/kubeedge/idempotency_record` file to make sure only one task(config update or node upgrade) can be performed at a time  
+- Back up `edgecore.db, edgecore.yaml` to backup path `/etc/kubeedge/backup/update-config` 
+- Update the `edgecore.yaml` and restart edgecore.service 
+- rollback: if update failed, keadm will do rollback operation to start edgecore with origin configuration.
```

Comment:
if one of the thousand edgenodes happened to fail eventually, what would happen? it looks like this job failed gracefully and rollback all requested node after all?

Reply:
Only the nodes that fail to be updated will be rolled back. The proposal may not be described clearly, I will optimize it again.